packages/webapp/src/stories/DateRangeSelector/DateRangeSelector.stories.jsx

```diff
@@ -0,0 +1,172 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import { Suspense } from 'react';
+import { within, userEvent, screen, fireEvent } from '@storybook/testing-library';
+import { expect } from '@storybook/jest';
+import selectEvent from 'react-select-event';
+import moment from 'moment';
+import DateRangeSelector from '../../components/DateRangeSelector';
+import { componentDecorators } from '../Pages/config/Decorators';
+import { dateRangeOptions } from '../../components/DateRangeSelector/constants';
+import { FROM_DATE, TO_DATE } from '../../components/Form/DateRangePicker';
+
+export default {
+  title: 'Components/DateRangeSelector',
+  component: DateRangeSelector,
+  decorators: componentDecorators,
+};
+
+const commonArgs = {
+  onChangeDate: () => ({}),
+  changeDateMethod: () => ({}),
+};
+
+export const WithPlaceholder = {
+  args: {
+    ...commonArgs,
+    placeholder: 'Select Date Range',
+  },
+};
+
+export const WithDefaultOption = {
+  render: () => {
+    return (
+      <Suspense fallback={'Loading...'}>
+        <DateRangeSelector
+          {...commonArgs}
+          defaultDateRangeOptionValue={dateRangeOptions.YEAR_TO_DATE}
+        />
+        <div style={{ color: 'transparent' }}>Click Away Test</div>
+      </Suspense>
+    );
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+
+    let selectedOptionText = await canvas.findByText('Year to date');
+    expect(selectedOptionText).toBeInTheDocument();
+
+    const select = canvas.getByRole('combobox');
+    await selectEvent.openMenu(select);
+
+    let option = screen.getByText('Last 7 days');
+    await userEvent.click(option);
+
+    await selectEvent.openMenu(select);
+    option = screen.getAllByText('Last 7 days')[1];
+    expect(option).toHaveStyle('font-weight: 700');
+
+    option = screen.getByText('Pick a custom range');
+    await userEvent.click(option);
+
+    const backButton = await canvas.findByText('back');
+    expect(backButton).toBeInTheDocument();
+    expect(backButton).toHaveStyle('color: #d4dae3');
+
+    await userEvent.click(backButton);
+    expect(backButton).toBeInTheDocument();
+
+    selectedOptionText = canvas.getByText('yyyy-mm-dd - yyyy-mm-dd');
+    expect(selectedOptionText).toBeInTheDocument();
+
+    const [input1, input2] = canvas.getAllByTestId('input');
+    await userEvent.clear(input1);
+    await userEvent.clear(input2);
+
+    await userEvent.type(input1, '2023-11-01');
+    expect(selectedOptionText).toBeInTheDocument();
+
+    await userEvent.type(input2, '2022-11-01');
+    expect(selectedOptionText).toBeInTheDocument();
+
+    const errorMessage = await canvas.findByText(
+      `End date must be after start date to return results`,
+    );
+    expect(errorMessage).toBeInTheDocument();
+
+    // This does not pass when rerunning on storybook UI
```

Comment:
I get a failure at this step in both CLI (i.e. running `pnpm test-storybook DateRangeSelector`) and the Storybook UI. The failure seems legitimate because the placeholder `yyyy-mm-dd yyyy-mm-dd` is still shown after clicking away rather than 'Year to date' (until the dropdown is re-selected). Is that behaviour not correct?

Reply:
Me too ^^

Reply:
I guess it was mysteriously passing...
I realized that I should not have used `useSelector` in the component to make it work in storybook, so I fixed it. However, click away behaviour couldn't be tested still...! So I gave up and removed the test this time 😢 
Thank you so much for testing!

Reply:
Oh it still didn't pass for you? Because now the clickAway behaviour *is* working for me in the interactive Storybook 😆! It's the store slice being mocked state that did it right? But I'm fine with skipping that part of testing 🙂 Thanks for writing the interaction tests!

Reply:
Storybook doesn't click away for me for some reason...

I was doing `const dateRange = useSelector(dateRangeSelector)`, but `dateRange` was becoming `undefined` in storybook and the error (`undefined.option`) was happening in `onClickAway` function.
![Screenshot 2023-10-13 at 11 21 19 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/af03ccf0-7171-4111-8cc4-c4097e0d9a66)

Thank you 😄

Reply:
Yes, me too! Exactly like that! But not since you updated this part:

```js
// DateRangeSelectorStories.jsx

const [dateRange, setDateRange] = useState({ option: dateRangeOptions.YEAR_TO_DATE });
  
return (
  <Suspense fallback={'Loading...'}>
    <DateRangeSelector
        dateRange={dateRange}
        changeDateMethod={() => ({})}
        setDateRangeOptionValue={(option) => setDateRange({ option })}
 ```
 
Now `dateRange.option` is always defined correctly for me even in Storybook so the clickAway has begun happening. I thought that was a very clever solution!

Reply:
Thank you Joyce! ❤️